lib/storage/src/upload/Upload.ts

```diff
@@ -0,0 +1,85 @@
+import { PutObjectCommandInput, S3, S3Client, ServiceOutputTypes } from "@aws-sdk/client-s3";
+
+import { Upload as UploadType } from "./types";
+export { Upload as UploadType } from "./types";
+import { EventEmitter } from "events";
+
+import { DEFAULT } from "./defaults";
+import { S3ClientUploader } from "./service-clients/S3ClientUploader";
+import { S3Uploader } from "./service-clients/S3Uploader";
+import { Uploader } from "./service-clients/Uploader";
+
+export class Upload extends EventEmitter {
+  params: PutObjectCommandInput;
+  client: UploadType.ServiceClients;
+  configuration: UploadType.Configuration;
+  progress: (progress: UploadType.Progress) => void = () => {};
+  uploader: Uploader;
+  uploadEvent = "";
+
+  /**
+   * Creates a managed upload object with a set of configuration options.
+   */
+  constructor(options: UploadType.Options) {
+    super();
+
+    this.params = options.params;
+    this.client = options.client;
+
+    const defaultOptions: UploadType.Configuration = {
+      queueSize: DEFAULT.QUEUE_SIZE,
+      partSize: DEFAULT.MIN_PART_SIZE,
+      leavePartsOnError: DEFAULT.ABORT_ON_FAILURE,
```

Comment:
Actually `leavePartsOnError` is `!DEFAULT.ABORT_ON_FAILURE`. Change this OR the fix recommended on line 71

Reply:
yep added this fix in the appropriate areas. Nice catch.